scwidgets/converters.py

```diff
@@ -0,0 +1,77 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+# @file
+# @author Matthew Andrew
+
+import scipp as sc
+import pathlib
+
+scipp_object_mutable = (sc.DataArray, sc.Dataset, sc.Variable,
+                        sc.DataArrayView, sc.DatasetView, sc.VariableView)
+
+scipp_object = scipp_object_mutable + (
+    sc.DataArrayConstView, sc.DatasetConstView, sc.VariableConstView)
+
+
+def filepath_converter(filepath, data_directory=pathlib.Path.cwd()):
```

Comment:
Is `converter` the right nomenclature? Some do conversions, others do validations, others configure the inputs (e.g., drop-down options). Maybe `input` is a better name?